jplag.endToEndTesting/src/main/java/de/jplag/end_to_end_testing/helper/JPlagTestSuiteHelper.java

```diff
@@ -64,8 +76,120 @@ public TestCaseModel createNewTestCase(String[] classNames, String functionName)
      * @throws IOException if an I/O error occurs
      */
     public void clear() throws IOException {
-        logger.debug("Class instance was cleaned!");
         deleteCopiedFiles(new File(TestDirectoryConstants.TEMPORARY_SUBMISSION_DIRECTORY_NAME));
+        logger.debug("Class instance was cleaned!");
+    }
+
+    /**
+     * @return Path to the stored test results
+     */
+    public Path getResultJsonPath() {
+        return resultJsonPath;
+    }
+
+    /**
+     * @return all resources names for the current specified language
+     */
+    public String[] getAllTestFileNames() {
+        return resourceNames;
+    }
+
+    /**
+     * saves all temporary results for a test into a json file
+     * @param jplagComparisonList list of elements to be saved
+     * @throws IOException Signals that an I/O exception of some sort has occurred. Thisclass is the general class of
+     * exceptions produced by failed orinterrupted I/O operations.
+     * @throws NoSuchAlgorithmException when no hash algorithm could be found
+     * @throws NameNotFoundException if the no filenames coud be found in the JPlagCOmparison object
+     */
+    public void saveTemporaryResult(List<JPlagComparison> jplagComparisonList, String functionName)
+            throws IOException, NoSuchAlgorithmException, NameNotFoundException {
+        for (JPlagComparison jplagComparison : jplagComparisonList) {
+            JsonHelper.writeResultModelToJsonFile(new ResultModel(jplagComparison, getTestIdentifier(jplagComparison)),
+                    LanguageToPathMapper.getTemporaryResultPathFromLanguageOption(languageOption).toString(), functionName,
+                    getTemporaryFileNameForJson(jplagComparison));
+        }
+    }
+
+    /**
+     * Creates a unique identifier from the submissions in the JPlagComparison object which is used to find the results in
+     * the json files.
+     * @param jPlagComparison object from which the hash should be generated
+     * @return unique identifier for test case recognition
+     */
+    public String getTestIdentifier(JPlagComparison jPlagComparison) {
+
+        String testFileNamesInFirstSubmission = getEnclosedFileNamesFromCollection(jPlagComparison.getFirstSubmission().getFiles());
+        String testFileNamesInSecondSubmission = getEnclosedFileNamesFromCollection(jPlagComparison.getSecondSubmission().getFiles());
+
+        int compaire = testFileNamesInFirstSubmission.compareTo(testFileNamesInSecondSubmission);
+
+        String returnIdentifier;
+
+        if (compaire != 0) {
```

Comment:
File names should never be equal as we control the file names. Thus we can simply have `if (a < b) a + b else b + a`. Also I would suggest to introduce a separator between the file names for readability. Maybe a `-` as underscores are already used in the file names?

Reply:
Changed to:

``` java
/**
     * Creates a unique identifier from the submissions in the JPlagComparison object which is used to find the results in
     * the json files.
     * @param jPlagComparison object from which the hash should be generated
     * @return unique identifier for test case recognition
     */
    public static String getTestIdentifier(JPlagComparison jPlagComparison) {

        String testFileNamesInFirstSubmission = getEnclosedFileNamesFromCollection(jPlagComparison.getFirstSubmission().getFiles());
        String testFileNamesInSecondSubmission = getEnclosedFileNamesFromCollection(jPlagComparison.getSecondSubmission().getFiles());

        int compaire = testFileNamesInFirstSubmission.compareTo(testFileNamesInSecondSubmission);

        return compaire < 0 ? testFileNamesInFirstSubmission + "_" + testFileNamesInSecondSubmission
                : testFileNamesInSecondSubmission + "_" + testFileNamesInFirstSubmission;
    }
    
    /**
     * Merges all contained filenames together without extension
     * @param files whose names are to be merged
     * @return merged filenames
     */
    private static String getEnclosedFileNamesFromCollection(Collection<File> files) {
        StringBuilder stringBuilder = new StringBuilder(files.size());
        for (File file : files) {
            String fileName = file.getName();
            stringBuilder.append(fileName.substring(0, fileName.lastIndexOf('.')));
        }

        return stringBuilder.toString();
    }
```

Reply:
As said, I would go with "-" as "_" is already used in the file names itself.
Also it is `compare` not `compaire`